tests/agent-opentracing/src/test/java/org/hawkular/apm/tests/agent/opentracing/jms/JavaxJMSITest.java

```diff
@@ -0,0 +1,261 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageListener;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TemporaryQueue;
+import javax.jms.TextMessage;
+
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.hawkular.apm.api.model.trace.Trace;
+import org.hawkular.apm.tests.common.ClientTestBase;
+import org.hawkular.apm.tests.common.Wait;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author gbrown
+ */
+public class JavaxJMSITest extends ClientTestBase {
+
+    private static ConnectionFactory connectionFactory;
+
+    private static final String TEST_MESSAGE = "TestMessage";
+    private static final String TEST_QUEUE = "TestQueue";
+    private static final String TEST_TOPIC = "TestTopic";
+    private static final String SUFFIX = "Reply";
+
+    private static ExecutorService executorService;
+
+    @BeforeClass
+    public static void initClass() throws Exception {
+        executorService = Executors.newCachedThreadPool();
+
+        connectionFactory = new ActiveMQConnectionFactory("vm://localhost?broker.persistent=false");
+    }
+
+    @Test
+    public void testQueueRequestOnly() throws JMSException, InterruptedException, ExecutionException {
+        queueSend(false);
+    }
+
+    @Test
+    public void testQueueRequestResponse() throws JMSException, InterruptedException, ExecutionException {
+        queueSend(true);
+    }
+
+    protected void queueSend(boolean withResponse) throws JMSException, InterruptedException, ExecutionException {
+        Connection connection = connectionFactory.createConnection();
+        connection.start();
+
+        Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+        Future<String> serverResult = executorService.submit(new JMSQueueAsyncServer());
+
+        MessageProducer producer = session.createProducer(session.createQueue(TEST_QUEUE));
+
+        TextMessage request = session.createTextMessage(TEST_MESSAGE);
+
+        TemporaryQueue responseQueue = null;
+        if (withResponse) {
+            responseQueue = session.createTemporaryQueue();
+            request.setJMSReplyTo(responseQueue);
+        }
+
+        producer.send(request);
+
+        if (withResponse) {
```

Comment:
Actually this can be removed, as currently not dealing with temp topics/queues (which are used for responses). So this can be added back in (and tested better) when the rules deal with response patterns.